core/shared/src/main/scala/fs2/Chunk.scala

```diff
@@ -1298,5 +1446,137 @@ object Chunk
           fa: Chunk[A]
       )(f: A => F[Option[B]])(implicit F: Applicative[F]): F[Chunk[B]] = fa.traverseFilter(f)
       override def mapFilter[A, B](fa: Chunk[A])(f: A => Option[B]): Chunk[B] = fa.mapFilter(f)
+
+      override def contains_[A](fa: Chunk[A], v: A)(implicit ev: Eq[A]): Boolean =
+        fa.contains(v)
+
+      override def count[A](fa: Chunk[A])(p: A => Boolean): Long =
+        fa.count(p).toLong
+
+      override def exists[A](fa: Chunk[A])(p: A => Boolean): Boolean =
+        fa.exists(p)
+
+      override def forall[A](fa: Chunk[A])(p: A => Boolean): Boolean =
+        fa.forall(p)
     }
 }
+
+private[fs2] final class ChunkAsSeq[+O](
+    private[fs2] val chunk: Chunk[O]
+) extends IndexedSeq[O]
+    with ChunkAsSeqPlatform[O]
+    with Serializable {
+  override def iterator: Iterator[O] =
+    chunk.iterator
+
+  override def apply(i: Int): O =
+    chunk.apply(i)
+
+  override def length: Int =
+    chunk.size
+
+  override def isEmpty: Boolean =
+    chunk.isEmpty
+
+  override def reverseIterator: Iterator[O] =
+    chunk.reverseIterator
+
+  override def foreach[U](f: O => U): Unit =
+    chunk.foreach { o => f(o); () }
+
+  override def headOption: Option[O] =
+    chunk.head
+
+  override def head: O =
+    if (chunk.nonEmpty) chunk.apply(0)
+    else throw new NoSuchElementException("head of empty Seq")
+
+  override def lastOption: Option[O] =
+    chunk.last
+
+  override def last: O =
+    if (chunk.nonEmpty) chunk.apply(chunk.size - 1)
+    else throw new NoSuchElementException("tail of empty Seq")
+
+  override def filter(p: O => Boolean): IndexedSeq[O] =
+    new ChunkAsSeq(chunk.filter(p))
+
+  override def take(n: Int): IndexedSeq[O] =
+    new ChunkAsSeq(chunk.take(n))
+
+  override def takeRight(n: Int): IndexedSeq[O] =
+    new ChunkAsSeq(chunk.takeRight(n))
+
+  override def toArray[O2 >: O: ClassTag]: Array[O2] =
+    chunk.toArray
+
+  override def toList: List[O] =
+    chunk.toList
+
+  override def toVector: Vector[O] =
+    chunk.toVector
+
+  override def toString: String =
+    chunk.iterator.mkString("ChunkAsSeq(", ", ", ")")
+
+  override def hashCode: Int = {
+    import util.hashing.MurmurHash3
+    var h = MurmurHash3.stringHash("ChunkAsSeq")
```

Comment:
Not sure if we should mix the class name (`ChunkAsSeq`) into the hash code, based on the `equals` implementation. For example, if the `ChunkAsSeq` instance is equal to the `GSeq` (e.g. coming from the `IndexedSeqChunk`), then they should have the same hash code values. At least that's my understanding, hope I haven't mixed anything. Maybe computing it only from the `chunk` contents would be more appropriate. :slightly_smiling_face:

Reply:
Man, the collections stdlib are a nightmare to understand.
Thanks for the heads up, just fixed it _(turned out, we shouldn't have need to override this in the first place)_.